gen/calc.py

```diff
@@ -920,6 +972,11 @@ def calculate_fault_domain_detect_contents(fault_domain_detect_filename):
         'weights': '',
         'adminrouter_auth_enabled': calculate_adminrouter_auth_enabled,
         'adminrouter_tls_1_0_enabled': 'false',
+        'adminrouter_tls_1_1_enabled': 'true',
+        'adminrouter_tls_1_2_enabled': 'true',
+        'adminrouter_tls_version_override': calculate_adminrouter_tls_version_override,
+        'adminrouter_tls_cipher_override': calculate_adminrouter_tls_cipher_override,
+        'adminrouter_tls_cipher_suite': '',
```

Comment:
For documentation it is important to note that both `_override` options are fully internal, i.e. they are not supposed to be set by operators in the config YAML document. Correct?

Reply:
technically the TLS version override could be overwritten by the user (I did this in case someone needed TLSv1.3 at some point so we'd have a workaround before building a formal interface).  The other one shouldn't ever be set by the user.

Reply:
Ah, ok, thanks for clarifying the intent.

Having both, `adminrouter_tls_1_x_enabled` as well as `adminrouter_tls_version_override` as public configuration variables seems to create unnecessary complexity.

Especially in configuration interfaces there should be one way to achieve a given thing. Unnecessary complexity is not so good in terms of using, testing, reasoning, documentation.

I think we have (had) two choices:
1) Use `adminrouter_tls_1_x_enabled`, and once we support TLS 1.Y we can introduce `adminrouter_tls_1_Y_enabled` (rare event).
2) Deprecate `adminrouter_tls_1_0_enabled` and have *only* a single configuration parameter for the entire version specification string.

A mixture of both I think is not healthy.

Can we think of a use case where a mixture enables something that (1) or (2) alone would not enable?

Reply:
I've decided to go with option (1) and remove `adminrouter_tls_version_override` configuration option.